Linking number calculation

kaokitich opened the pull request:
Kao's summer project

Commit:
shutting down save

Commit:
Restructure Lk calc
- Lk, Tw, Wr now are integrated values
- LkScratch, TwScratch, WrScratch are only for the whole-ring "from 
scratch" calculation so that we can compare them to the integrated 
values in verifyEnergiesFromScratch
- initialize Lk,Tw, Wr in calculateEnergyFromScratch in 
wlcsim_quinn.onlyNode

Commit:
merge savepointLinkingNumber to MC_linkingNumber

Commit:
Linking number working

qmacAstanford commented on src/mc/MC_crank.f03:
```diff
@@ -49,11 +51,8 @@ subroutine MC_crank(IB1,IB2,IT1,IT2,MCAMP,WindoW,rand_stat,dib,success)
-!TOdo saving RP is not actually needed, even in these cases, but Brad's code assumes that we have RP.
-if (WLC_P__RING .OR.WLC_P__INTERP_BEAD_LENNARD_JONES) then
+if (WLC_P__NO_SELF_CROSSING) then
     wlc_RP = wlc_R
```
Slow

qmacAstanford commented on src/mc/MC_writhe.f03:
```diff
@@ -1,6 +1,6 @@
   !Calculate the writhe between segment that was moved and segment that was not moved using the method of Klenin (2000)
 
-subroutine WRITHE_MOVE(R,RP,IB1,IB2,IT1,IT2,IP,N,Wr)
+subroutine WRITHE_MOVE(R,RP,IB1,IB2,IT1,IT2,N,Wr)
```
Check to see if used elsewhere and change

qmacAstanford commented on src/mc/MCsim.f03:
```diff
@@ -51,7 +52,7 @@ pure function list_confinement()
     logical forward           ! direction of reptation move
 
-    integer I,J
+    integer I,J, I_left
```
I this used

qmacAstanford commented on src/mc/MCsim.f03:
```diff
@@ -61,7 +62,7 @@ pure function list_confinement()
     real(dp) VHC      ! HC strength
 !    real(dp) phiTot  ! for testing
-    real(dp) wrp ! proposed writhe
+!    real(dp) wrp ! proposed writhe
```
Actually delete

qmacAstanford commented on src/mc/MCsim.f03:
```diff
@@ -240,6 +247,19 @@ pure function list_confinement()
               call umbrella_energy()
           endif
 
+          if (WLC_P__NO_SELF_CROSSING) then
```
Please comment

qmacAstanford commented on src/mc/MCsim.f03:
```diff
@@ -240,6 +247,19 @@ pure function list_confinement()
+              LkP = wlc_Lk + delLk
+              if (abs(LkP - wlc_Lk0) > WLC_P__NO_CROSSING_CUTOFF) then
+!                  print*, 'wlc_LkMove when skip', wlc_LkMove
+!                  print*, 'skip move because linking nubmer changes'
```
Delete these prints

qmacAstanford commented on src/wlcsim/initcond.f03:
```diff
@@ -492,27 +492,35 @@ subroutine wlc_init(R, U, NB, EPS, l0, rand_stat)
-
          call random_number(urand,rand_stat)
          theta = urand(1)*2.0_dp*pi
          z = (1.0_dp/EPS)*log(2.0_dp*sinh(EPS)*urand(2)+exp(-EPS))
```
Revert to old for entire file

qmacAstanford commented on src/wlcsim/params.f03:
```diff
@@ -774,7 +787,8 @@ subroutine initialize_wlcsim_data( wlc_p)
             call constructBin(wlc_bin,setBinShape,setMinXYZ,setBinSize)
             do i=1,WLC_P__NT
-                if ((WLC_P__CONFINETYPE == 'excludedShpereInPeriodic')&
+                if (WLC_P__NEIGHBOR_BINS .and.&
```
Revert

qmacAstanford reviewed (changes requested):
A few small changes

Commit:
Minor fixes
- delete copying wlc_R to wlc_RP in MC_crank
- delete comments & unused variables in MCsim
- replace initcond and params

Commit:
Add parameters for linking number calculation in defines

Commit:
Add idiot check for no self crossing
